tests/attenuation.test.ts

```diff
@@ -306,4 +310,156 @@ describe("hasCapability", () => {
     expect(cap).toEqual(false)
   })
 
+  it("supports redelegation via forwarding with a `my` capability", async () => {
+    // alice -> bob
+    // alice delegates access to sending email as her to bob
+    // and bob delegates it further to mallory
+    const leafUcan = await token.build({
+      issuer: alice,
+      audience: bob.did(),
+      capabilities: [ emailCapability("alice@email.com") ]
+    })
+
+    const ucan = await token.build({
+      issuer: bob,
+      audience: mallory.did(),
+      capabilities: [ capability.my(capability.superUser.SUPERUSER) ],
+      proofs: [ token.encode(leafUcan) ]
+    })
```

Comment:
I think that's not quite how the `my` capability is supposed to work, I would actually expect the test to fail according to the spec.

Here, the `my` shouldn't give access to all capabilities in the proofs (the `alice@email.com` cap in this case), that's what the `prf:*` or `prf:0` resources are supposed to do. If `my` got replaced with the `prf` capability, I'd expect the test to succeed again.

The way I'd expect the `my` capability to be tested here would be if we literally just swapped the `capabilities` arrays between these two UCAN payloads. Then Alice gives access to all her capabilities to Bob, and bob can delegate only the email access further.

Reply:
> I assume I can't really test if Bob, or Mallory, can perform a specific capability, because the my lets them do anything. But I can test if Alice is the originator?

Yeah exactly, `my` and `as` is all about the originator DID.

> "bob can delegate only the email access further" doesn't really make sense to me here. If Alice delegated everything to Bob, can't Bob re-delegate everything to Mallory?

Sorry, that's a confusing sentence. "bob can delegate only" vs. "bob can only delegate".
Maybe I should've written "then it's possible for to delegate email access further, without delegating everything else".

Reply:
![](https://media.giphy.com/media/rHR8qP1mC5V3G/giphy.gif)